core/src/main/java/com/oath/oak/OakAttachedWriteBuffer.java

```diff
@@ -34,56 +71,57 @@ void validateAccess() {
     @Override
     public OakWriteBuffer put(int index, byte value) {
         validateAccess();
-        getDataByteBuffer().put(getDataOffset(index), value);
+        writeBuffer.put(getDataOffset(index), value);
         return this;
     }
 
     @Override
     public OakWriteBuffer putChar(int index, char value) {
         validateAccess();
-        getDataByteBuffer().putChar(getDataOffset(index), value);
+        writeBuffer.putChar(getDataOffset(index), value);
         return this;
     }
 
     @Override
     public OakWriteBuffer putShort(int index, short value) {
         validateAccess();
-        getDataByteBuffer().putShort(getDataOffset(index), value);
+        writeBuffer.putShort(getDataOffset(index), value);
         return this;
     }
 
     @Override
     public OakWriteBuffer putInt(int index, int value) {
         validateAccess();
-        getDataByteBuffer().putInt(getDataOffset(index), value);
+        writeBuffer.putInt(getDataOffset(index), value);
         return this;
     }
 
     @Override
     public OakWriteBuffer putLong(int index, long value) {
         validateAccess();
-        getDataByteBuffer().putLong(getDataOffset(index), value);
+        writeBuffer.putLong(getDataOffset(index), value);
         return this;
     }
 
     @Override
     public OakWriteBuffer putFloat(int index, float value) {
         validateAccess();
-        getDataByteBuffer().putFloat(getDataOffset(index), value);
+        writeBuffer.putFloat(getDataOffset(index), value);
         return this;
     }
 
     @Override
     public OakWriteBuffer putDouble(int index, double value) {
         validateAccess();
-        getDataByteBuffer().putDouble(getDataOffset(index), value);
+        writeBuffer.putDouble(getDataOffset(index), value);
         return this;
     }
 
     /*-------------- OakUnsafeDirectBuffer --------------*/
```

Comment:
I am may be confused, where getAddress/getOffset are implemented?

Reply:
They are implemented in `Slice`